src/main/java/org/testng/TestNG.java

```diff
@@ -416,6 +416,15 @@ private void initProcessor(Parser result) {
    * original suite.
    */
   private static XmlSuite extractTestNames(XmlSuite s, List<String> testNames) {
+    List<XmlSuite> childSuites = s.getChildSuites();
```

Comment:
Can you add a comment about what you're trying to do here? The recursion is not immediately apparent to me.